sklearn/semi_supervised/tests/test_self_training.py

```diff
@@ -285,6 +290,7 @@ def test_k_best_selects_best():
         assert row in added_by_st
 
 
+@pytest.mark.filterwarnings("ignore::FutureWarning")
```

Comment:
Here we can probably use clf_1 and clf_2 set to `LogisticRegression()` in the first half of the test and use `svc_1` and `svc_2`  set to `SVC()` (with the implicit default value for the `probability` parameter) instead of hiding the warning.